### PR TITLE
Limit the number of threads for R example

### PR DESCRIPTION
**Pull request by anjalirai-intel**
With the latest Patched R version 4.2.1, In centos server it is allocating memory to each CPU and because of this Gramine has ran out of memory, To avoid that limiting the number of threads to 4.

Signed-off-by: Anjali Rai <anjali.rai@intel.com>

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)

### CODE REVIEW

**Comment by anjalirai-intel** [comment]
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/gramineproject/examples/40#-NCPGZP60Pi7NW56mCWT):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I would rephrase like this:
```
The latest version of R (Patched 4.2.1) uses OpenBLAS with OpenMP,
which allocates 128MB of memory for each potentially created thread.
Since by default the number of threads is calculated by OpenBLAS as
the number of available CPU cores, this per-thread memory allocation
may quickly exceed the enclave size on large platforms.

To avoid running out of enclave memory, we set OMP_NUM_THREADS
environment variable to 4, to limit the number of created threads.

This was detected on a CentOS 8 platform with a large number of CPU
cores (Gramine ran out of enclave memory).
```
</blockquote></details>

Done.

**Comment by anjalirai-intel** [comment]
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/examples/40#-NCPGWQN9M9ITYyrUOAl):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`Limited` -> `Limit`
</blockquote></details>

Done.

**Review by dimakuv — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @anjalirai-intel)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/examples/40#-NCq4yPM0_wBTPGWyAW9):*




_Suggestion:_
```
Limit the number of threads for R example
```
___
*[`r/R.manifest.template` line 15 at r2](https://reviewable.io/reviews/gramineproject/examples/40#-NCq5Dcm-PJf_6lmKohZ:-NCq5Dcm-PJf_6lmKoh_:biy4sy5) ([raw file](https://github.com/gramineproject/examples/blob/4855c519b86e4dbe114fa5eaadbdfe970a967a12/r/R.manifest.template#L15)):*
> ```
> loader.env.R_ARCH = ""
> loader.env.R_HOME = "{{ r_home }}"
> ```

Why have you removed the empty line?

**Review by dimakuv — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @mkow)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/examples/40#-NCq4yPM0_wBTPGWyAW9):*
I will fix it during final rebase.
___
*[`r/R.manifest.template` line 15 at r2](https://reviewable.io/reviews/gramineproject/examples/40#-NCq5Dcm-PJf_6lmKohZ:-NCsWk2g1kAf4v9CLmMj:b-t5fs2v) ([raw file](https://github.com/gramineproject/examples/blob/4855c519b86e4dbe114fa5eaadbdfe970a967a12/r/R.manifest.template#L15)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Why have you removed the empty line?
</blockquote></details>

I will fix it during final rebase.

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @mkow)

___
*[`-- commits` line 2 at r2](https://reviewable.io/reviews/gramineproject/examples/40#-NCq4yPM0_wBTPGWyAW9):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I will fix it during final rebase.
</blockquote></details>

Done
___
*[`r/R.manifest.template` line 15 at r2](https://reviewable.io/reviews/gramineproject/examples/40#-NCq5Dcm-PJf_6lmKohZ:-NCsctmp3DN47R7y2DiP:bb6un1j) ([raw file](https://github.com/gramineproject/examples/blob/4855c519b86e4dbe114fa5eaadbdfe970a967a12/r/R.manifest.template#L15)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I will fix it during final rebase.
</blockquote></details>

Done

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @anjalirai-intel)

___
*[`r/R.manifest.template` line 15 at r4](https://reviewable.io/reviews/gramineproject/examples/40#-NCtAXvg-zSNXKIJzEGI:-NCtAXvg-zSNXKIJzEGJ:b-x06n6c) ([raw file](https://github.com/gramineproject/examples/blob/aec332446af81e42615d142dbda45e2146d2f6c9/r/R.manifest.template#L15)):*
> ```
> loader.env.R_ARCH = ""
> loader.env.R_HOME = "{{ r_home }}"
> loader.env.OMP_NUM_THREADS="4"
> ```

Actually, there are spaces missing around `=`, could you quickly fix and rebase again?

**Comment by anjalirai-intel** [comment]
*[`r/R.manifest.template` line 15 at r4](https://reviewable.io/reviews/gramineproject/examples/40#-NCtAXvg-zSNXKIJzEGI:-NCtBv1O1SI67wmIcOAW:b44x69m) ([raw file](https://github.com/gramineproject/examples/blob/aec332446af81e42615d142dbda45e2146d2f6c9/r/R.manifest.template#L15)):*
> Sure
Done

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @anjalirai-intel)

___
*[`-- commits` line 2 at r5](https://reviewable.io/reviews/gramineproject/examples/40#-NCtCsB87OiX9aTPmL_l):*
umm, you just undid Dmitrii's change...

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @anjalirai-intel)

___
*[`-- commits` line 2 at r5](https://reviewable.io/reviews/gramineproject/examples/40#-NCtCsB87OiX9aTPmL_l):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

umm, you just undid Dmitrii's change...
</blockquote></details>

@anjalirai-intel Please do not touch this PR. We're in the process of final rebase and merge.

**Comment by anjalirai-intel** [comment]
*[`-- commits` line 2 at r5](https://reviewable.io/reviews/gramineproject/examples/40#-NCtCsB87OiX9aTPmL_l):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@anjalirai-intel Please do not touch this PR. We're in the process of final rebase and merge.
</blockquote></details>

ok

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @mkow)

___
*[`-- commits` line 2 at r5](https://reviewable.io/reviews/gramineproject/examples/40#-NCtCsB87OiX9aTPmL_l):*
<details><summary><i>Previously, anjalirai-intel wrote…</i></summary><blockquote>

ok
</blockquote></details>

Done @mkow

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/40)* status: :shipit: complete! all files reviewed, all discussions resolved
